.github/workflows/publish_release.yml

```diff
@@ -0,0 +1,170 @@
+name: Publish NovelRT
+
+on:
+  workflow_dispatch:
+
+jobs:
+  create-release:
+    runs-on: ubuntu-latest
+
+    outputs:
+      tag: ${{ steps.tag_version.outputs.new_tag }}
+      id: ${{ steps.create_release.outputs.id }}
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - uses: mathieudutour/github-tag-action@v6.0
+      id: tag_version
+      with:
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+
+    - uses: ncipollo/release-action@v1
+      id: create_release
+      with:
+        tag: ${{ steps.tag_version.outputs.new_tag }}
+        name: NovelRT ${{ steps.tag_version.outputs.new_tag }}
+        body: ${{ steps.tag_version.outputs.changelog }}
+        prerelease: true
+        draft: true
+
+  build-release:
+    needs: create-release
+    runs-on: ${{ matrix.os }}
+    container: ${{ matrix.container }}
+
+    strategy:
+      matrix:
+        configuration: [ RelWithDebInfo, MinSizeRel ]
+        os: [ ubuntu-latest, windows-latest, macos-10.15 ]
+        exclude:
+        - os: macos-10.15
+          configuration: RelWithDebInfo
+        - os: macos-10.15
+          configuration: MinSizeRel
+        include:
+        - os: ubuntu-latest
+          container: novelrt/novelrt-build:latest
+          profile: linux-clang10-amd64
+          name: Ubuntu
+        - os: macos-10.15
+          profile: macos-appleclang12-amd64
+          configuration: Release
+          name: macOS
+        - os: macos-10.15
+          profile: macos-appleclang12-amd64
+          configuration: Debug
+          name: macOS
+        - os: windows-latest
+          profile: windows-vs2022-amd64
+          name: Windows
+    steps:
+    - uses: actions/checkout@v2
+    - run: git fetch --depth=1 origin +refs/tags/*:refs/tags/*
```

Comment:
This is only necessary for the gitversion stuff I have in Fabulist - if we're not using it then we can ignore it